simde/x86/sse4.2.h

```diff
@@ -37,6 +37,204 @@ SIMDE__BEGIN_DECLS
 #  define SIMDE_X86_SSE4_2_ENABLE_NATIVE_ALIASES
 #endif
 
+SIMDE__FUNCTION_ATTRIBUTES
+int
+simde_mm_cmpestra(simde__m128i a, int la, simde__m128i b, int lb, const int imm8){
+#if defined(SIMDE_X86_SSE4_2_NATIVE)
+  return _mm_cmpestra(a, la, b, lb, imm8);
+#else
+  simde__m128i_private
+  BoolRes_ = simde__m128i_to_private(simde_mm_setzero_si128()),
+  a_ = simde__m128i_to_private(a),
+  b_ = simde__m128i_to_private(b);
+  if(imm8&1){
```

Comment:
> This should have spaces on either side of the and operator to match the coding style in SIMDe. `imm8 & 1` is much easier to read. There are a lot of these in this code, so instead of pointing them out individually I'll just say that binary operators should have spacing around them, parenthesis should not (unless the spaces clarify something or make code line up, of course).

I have made changes accordingly.

> I also suggest pulling out the different bits out of imm8 and put them in their own variables, which describe what they're for

I have resolved all other suggestions here hopefully.

Reply:
> > This should have spaces on either side of the and operator to match the coding style in SIMDe. `imm8 & 1` is much easier to read. There are a lot of these in this code, so instead of pointing them out individually I'll just say that binary operators should have spacing around them, parenthesis should not (unless the spaces clarify something or make code line up, of course).
> 

It seemed to me that, in the definition of SIMDe functions in existing code, there was generally a space between the function name and the parenthesis opening its arguments.
I therefore tried to follow this style (which is contrary to my normal custom).
Is a space preferred or not?

While on the subject of coding style:
In the opener of for loops you put a space before the semi-colons.
Again, while I generally never do this, I try to follow the existing style.
Is this correct?
Also in for loop openers there are nearly always a `(sizeof(array) / sizeof(array[0]))`.
Why not define a macro: `LENGTHOF(array)` or `ARRAY_LENGTH(array)` or whatever for this?

Reply:
> Is a space preferred or not?

I generally like the space since I think it makes the code a bit more readable, but it's not a big deal.  You can't put a space before the argument list on a function-like macro anyways.

> Is this correct?

You mean `for(int i = 0 ; i < x ; i++)` vs. `for (int i = 0;i < x; i++)`?  Yes, the former.  In general, spaces around operators make the code more readable.

> Why not define a macro

Because I find it a bit too magical and it's hard to understand exactly what is happening without finding the macro definition.  Also, a macro like that wouldn't work for pointers (including parameters), and again it's not obvious why unless you look up the macro.

In general, unless we really need a macro for something (usually to choose between different implementations using the preprocessor) I try to avoid macros, especially when it's not obvious what they do and when they'll work.